src/services/services.ts

```diff
@@ -3705,6 +3732,41 @@ namespace ts {
                 return entries;
             }
 
+            function getJsDocCompletionEntries(fileName: string, position: number): CompletionEntry[] {
+                let sourceFile = getValidSourceFile(fileName);
+                let hasJsDocComment = ts.hasDocComment(sourceFile, position);
+                if (!hasJsDocComment) {
+                    return undefined;
+                }
+
+                // The current position is right after an At sign
+                if (sourceFile.text.charCodeAt(position - 1) === CharacterCodes.at) {
+                    return getAllJsDocCompletionEntries();
+                }
+
+                // Or if the current position is in a tag name
+                let tag = getJsDocTagAtPosition(sourceFile, position);
+                if (tag) {
+                    if (position >= tag.atToken.end && position <= tag.tagName.end) {
+                        return getAllJsDocCompletionEntries();
+                    }
+                }
+            }
+
+            function getAllJsDocCompletionEntries(): CompletionEntry[] {
+                let tagNames = ["augments", "author", "argument", "borrows", "class", "constant", "constructor", "constructs", "default", "deprecated", "description", "event", "example", "extends", "field", "fileOverview", "function", "ignore", "inner", "lends", "link", "memberOf", "name", "namespace", "param", "private", "property", "public", "requires", "returns", "see", "since", "static", "throws", "type", "version"];

```

Comment:
This must be stored somewhere else as well.

Please
1. Make this a const
2. Move it out of the function so this list isn't created upon every invocation
3. Break it into multiple lines.